### PR TITLE
Layout for MS Either Neither

### PR DESCRIPTION
**Pull request by beausmith**
[east-weir-flag-plus-either-neither.pdf](https://github.com/votingworks/election-manager/files/5128100/east-weir-flag-plus-either-neither.pdf)

Resolves: https://github.com/votingworks/vxmail/issues/143

### CODE REVIEW

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 1]
```diff
@@ -769,121 +817,199 @@ const HandMarkedPaperBallot = ({
...
-                        </React.Fragment>
+                          </React.Fragment>
+                        )}
+                      </p>
+                      <Text
+                        small
+                        preLine
+                        dangerouslySetInnerHTML={{
+                          __html: contest.description,
+                        }}
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:905]: We're not sanitizing this at all?

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 2]
```diff
@@ -769,121 +817,199 @@ const HandMarkedPaperBallot = ({
...
+            <OtherContestsLayout>
+              {otherContests.map((contest) => (
+                <Contest
+                  key={contest.id}
+                  data-contest
+                  data-contest-title={contest.title}
+                  section={dualPhraseWithSlash(
+                    contest.section,
+                    localeContestsById &&
+                      localeContestsById[contest.id]?.section,
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:870]:
```suggestion
                    localeContestsById?.[contest.id]?.section,
```

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 3]
```diff
@@ -769,121 +817,199 @@ const HandMarkedPaperBallot = ({
...
+                    { normal: true }
+                  )}
+                  title={dualPhraseWithSlash(
+                    contest.title,
+                    localeContestsById && localeContestsById[contest.id]?.title,
```
Comment (eventualbuddha) [inline client/src/components/HandMarkedPaperBallot.tsx:875]:
```suggestion
                    localeContestsById?.[contest.id]?.title,
```

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 4]
```diff
@@ -769,121 +817,199 @@ const HandMarkedPaperBallot = ({
...
-                            parties={parties}
-                            vote={votes[contest.id] as CandidateVote}
-                            locales={locales}
-                          />
-                        </React.Fragment>
-                      )}
-                      {contest.type === 'yesno' && (
-                        <React.Fragment>
-                          <p>
+            {candidateContests.map((contest) => (
```
[the remark this answers is not in the document]
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:828]: The default value for `candidateContests` is `[]` and thus there may be nothing to map over. So why is this length check necessary?

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 5]
```diff
@@ -769,121 +817,199 @@ const HandMarkedPaperBallot = ({
...
+              <Contest
+                key={contest.id}
+                data-contest
+                data-contest-title={contest.title}
```
[the remark this answers is not in the document]
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:832]: Do I recall @benadida adding them for some reason? Doesn't seem like they are necessary any longer from searching the code. I'm going to proactively remove these data-attributes.

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 1 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:912]: Not yet!

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (benadida) [reply client/src/components/HandMarkedPaperBallot.tsx:832]: I don't remember. Remove if it doesn't break anything.

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 2 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx]: done

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 1 above)
Comment (benadida) [reply client/src/components/HandMarkedPaperBallot.tsx:912]: there's a ticket for it:
https://github.com/votingworks/vxmail/issues/143

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 3 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx]: done

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply client/src/components/HandMarkedPaperBallot.tsx:828]: I was thinking "what happens if there are no candidate contests?", but I think there'd need to be bigger changes than this to support that.

[the remark this answers is not in the document]
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:828]: True. This is out of scope for M10. :)

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 1 above)
Comment (beausmith) [reply client/src/components/HandMarkedPaperBallot.tsx:910]: Adding this now.

**Pull request merged**